openpype/plugins/publish/extract_thumbnail.py

```diff
@@ -22,13 +22,40 @@ class ExtractThumbnail(pyblish.api.InstancePlugin):
         "imagesequence", "render", "render2d", "prerender",
         "source", "clip", "take", "online", "image"
     ]
-    hosts = ["shell", "fusion", "resolve", "traypublisher", "substancepainter"]
+    hosts = [
+        "shell",
+        "fusion",
+        "resolve",
+        "traypublisher",
+        "substancepainter",
+        "nuke",
+    ]
     enabled = False
 
-    # presetable attribute
+    # presentable attribute
     ffmpeg_args = None
 
     def process(self, instance):
+        # run main process
+        self._main_process(instance)
+
+        # Make sure cleanup happens to representations which are having both
+        # tags `delete` and `need_thumbnail`
+        for repre in tuple(instance.data["representations"]):
```

Comment:
why the cast to **tuple** is needed?

Reply:
It needs to make a 'copy' of the list since it's removing entries in that list as it iterates.